### ISSUE TITLE
Git ship panics when the parent of your branch is a perennial branch

### ISSUE THREAD

**Issue by Zekka** [Originate/git-town#972]
Conditions to reproduce:

1. Create a repository.
1. Create two branches, one called `master` and one called `develop` (or something like that)
1. Using `git town config --setup`, make `master` the main branch and `develop` a perennial branch.
1. Create a third branch using `git town hack some-feature` from `develop`
1. Create a file and commit that file to the new branch.
1. Ship that branch: `git town ship`

Expected behavior:
- git town will squash and rebase into `develop`

Actual behavior:
- git town panics.

Here is a sample repo configuration that creates this problem:

```
$ git town config
Main branch:
  master

Perennial branches:
  perennial-branch

Branch Ancestry:
  master

Pull branch strategy:
  rebase

git-hack push flag:
  false

$ git branch
* feature-branch
  master
  perennial-branch
```

Here is the output from git town ship:
```
$ git town ship
[feature-branch] git fetch --prune
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/Originate/git-town/src/cmd.ensureParentBranchIsMainBranch(0xc42000b640, 0x1a)
        /home/travis/gopath/src/github.com/Originate/git-town/src/cmd/ship.go:89 +0x248
github.com/Originate/git-town/src/cmd.checkShipPreconditions(0x13635a8, 0x0, 0x0, 0xc42000b640, 0x1a, 0x0, 0xc42000b640, 0x1a)
        /home/travis/gopath/src/github.com/Originate/git-town/src/cmd/ship.go:81 +0x199
github.com/Originate/git-town/src/cmd.glob..func33.3(0x121e269, 0x4, 0xc4200001a0)
        /home/travis/gopath/src/github.com/Originate/git-town/src/cmd/ship.go:53 +0x40
github.com/Originate/git-town/src/steps.Run(0x122e340, 0x121e269, 0x4, 0x0, 0x122e348, 0xc42000aec0)
        /home/travis/gopath/src/github.com/Originate/git-town/src/steps/run.go:54 +0x38d
github.com/Originate/git-town/src/cmd.glob..func33(0x1345ea0, 0x13635a8, 0x0, 0x0)
        /home/travis/gopath/src/github.com/Originate/git-town/src/cmd/ship.go:56 +0xfe
github.com/Originate/git-town/vendor/github.com/spf13/cobra.(*Command).execute(0x1345ea0, 0x13635a8, 0x0, 0x0, 0x1345ea0, 0x13635a8)
        /home/travis/gopath/src/github.com/Originate/git-town/vendor/github.com/spf13/cobra/command.go:648 +0x231
github.com/Originate/git-town/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1346d80, 0xc42006e001, 0x1226d53, 0x25)
        /home/travis/gopath/src/github.com/Originate/git-town/vendor/github.com/spf13/cobra/command.go:735 +0x339
github.com/Originate/git-town/vendor/github.com/spf13/cobra.(*Command).Execute(0x1346d80, 0x1346d80, 0xc42003ff50)
        /home/travis/gopath/src/github.com/Originate/git-town/vendor/github.com/spf13/cobra/command.go:693 +0x2b
github.com/Originate/git-town/src/cmd.Execute()
        /home/travis/gopath/src/github.com/Originate/git-town/src/cmd/root.go:27 +0x3d
main.main()
        /home/travis/gopath/src/github.com/Originate/git-town/main.go:6 +0x20
```

**Comment by kevgo** [comment]
Hey @Zekka, thanks for the bug report! Sorry for the late reply, I was on a business trip the last 3 weeks. I have created #993 to address this missing feature. Please take a look and tell me if the feature spec correctly reflects what you have in mind here. Thanks!  😸

**Comment by Zekka** [comment]
This would have fixed my problem at time of submitting bug report. Looks good to me!

**Comment by charlierudolph** [comment]
Resolved in #993 and released in 4.2.0
